pkg/deployments/tasks/20230314-batch-relayer-v5/test/composableStableV1.fork.ts

```diff
@@ -0,0 +1,184 @@
+    });
+
+    before('run composable stable pool task', async () => {
+      stableTask = new Task('20220906-composable-stable-pool', TaskMode.READ_ONLY, getForkedNetwork(hre));
+      factory = await stableTask.deployedInstance('ComposableStablePoolFactory');
+    });
+
+    before('deploy composable stable pool', async () => {
+      const tx = await factory.create(
+        'SP',
+        'SPT',
+        tokens,
+        amplificationParameter,
+        rateProviders,
+        cacheDurations,
+        exemptFlags,
+        swapFeePercentage,
+        owner.address
+      );
+      const event = expectEvent.inReceipt(await tx.wait(), 'PoolCreated');
+
+      pool = await stableTask.instanceAt('ComposableStablePool', event.args.pool);
+      expect(await factory.isPoolFromFactory(pool.address)).to.be.true;
+
+      poolId = await pool.getPoolId();
+      const [registeredAddress] = await vault.getPool(poolId);
+      expect(registeredAddress).to.equal(pool.address);
+    });
+
+    before('initialize composable stable pool', async () => {
+      const bptIndex = await pool.getBptIndex();
+
+      const composableInitialBalances = Array.from({ length: tokens.length + 1 }).map((_, i) =>
+        i == bptIndex ? 0 : i < bptIndex ? initialBalances[i] : initialBalances[i - 1]
+      );
+      const { tokens: allTokens } = await vault.getPoolTokens(poolId);
+
+      const userData = StablePoolEncoder.joinInit(composableInitialBalances);
+      await vault.connect(whale).joinPool(poolId, whale.address, owner.address, {
+        assets: allTokens,
+        maxAmountsIn: Array(tokens.length + 1).fill(MAX_UINT256),
+        fromInternalBalance: false,
+        userData,
+      });
+    });
+
+    // V1 does not support proportional exits
+    it('can exit with exact tokens through the relayer', async () => {
+      const bptBalance = await pool.balanceOf(owner.address);
+      expect(bptBalance).to.gt(0);
+
+      const vaultDAIBalanceBeforeExit = await dai.balanceOf(vault.address);
+      const ownerDAIBalanceBeforeExit = await dai.balanceOf(owner.address);
+      const { tokens: allTokens } = await vault.getPoolTokens(poolId);
+      const amountsOut = [fp(100), bn(100e6)];
+
+      const userData = defaultAbiCoder.encode(
+        ['uint256', 'uint256[]', 'uint256'],
+        [ComposableStablePoolV1ExitKind.BPT_IN_FOR_EXACT_TOKENS_OUT, amountsOut, MAX_UINT256]
+      );
+
+      /* The exit without the relayer
+      await vault.connect(owner).exitPool(poolId, owner.address, owner.address, {
+        assets: allTokens,
+        minAmountsOut: Array(tokens.length + 1).fill(0),
+        fromInternalBalance: false,
+        userData,
+      }); */
+
+      // Send BPT to the relayer so it can exit.
+      await pool.connect(owner).transfer(relayer.address, bptBalance);
+
+      const exitCalldata = library.interface.encodeFunctionData('exitPool', [
+        poolId,
+        PoolKind.COMPOSABLE_STABLE,
+        relayer.address,
+        owner.address,
```

Comment:
```suggestion
      const exitCalldata = library.interface.encodeFunctionData('exitPool', [
        poolId,
        PoolKind.COMPOSABLE_STABLE,
        owner.address,
        owner.address,
```

The whole point of the relayer is to have it use the user's funds. If the `sender` is the relayer, and we transfer the funds beforehand, then granting permission to the relayer etc doesn't do anything.